### PR TITLE
Use UUIDs as keys to operate with individuals

### PR DESCRIPTION
**Pull request by sduenas**
This PR updates some of the functions in the API to access `Individual` objects using their identities UUIDs.

So for example, if we have an `Individual` with three identities `AAAA`, `BBBB`, `CCCC` and we want to add a new identity linked to that individual any of the next calls will have the same effect:

```
add_identity(ctx, 'mysource', email='jsmith@example.com', uuid='AAAA')
add_identity(ctx, 'mysource', email='jsmith@example.com', uuid='BBBB')
add_identity(ctx, 'mysource', email='jsmith@example.com', uuid='CCCC')
```
The functions affected are `add_identity`, `update_profile`, `move_identity`, `lock`, `unlock`, `enroll`, `withdraw` and `merge_identities`.

This PR also introduces a new change in the model. The field `uuid` in the `Indentity` has been renamed to 'mk' (main_key) as UUIDs will only be a term used for identities;

This PR is related to #288 .

### CODE REVIEW

File: sortinghat/core/api.py [hunk 1]
```diff
@@ -959,7 +965,7 @@ def _delete_individuals(trxl, individuals):
     trxl = TransactionsLog.open('merge_identities', ctx)
 
     try:
-        to_individual = find_individual(to_uuid)
+        to_individual = find_individual_by_uuid(to_uuid)
         from_individuals = _find_individuals(from_uuids, to_uuid)
```
[the remark this answers is not in the document]
Comment (sduenas) [reply sortinghat/core/api.py:969]: I agree. It's more clear what you say.

**Comment by sduenas** [comment]
> @sduenas I have reviewed and tested this PR and everything worked as expected. Please, check my in-line comments.
> 
> I have one addtional comment: I think the help texts which are displayed in the command line client should be also updated with these changes in some parts, clarifying that any identity can be used to run the operation.
> 
> For instance, in the help text for `enroll` command it says: `UUID: individual to enroll`. I suggest to include here something like `UUID: identity from the individual to enroll` or something like that. WDYT? Thanks!

I understand what you mean. I'm having problems to manage the concept of uuid to make clear it any of the uuids related to an individual are valid. I was thinking to rename this parameters to `key` or something like that, but I'm not sure yet. If you don't mind, we can keep it as is now and think what works better.

**Comment by sduenas** [comment]
@mafesan please check my changes

**Comment by mafesan** [comment]
@sduenas I have reviewed the latest changes and everything LGTM. Thank you!

**Pull request merged**